Dockerfile

```diff
@@ -1,93 +1,46 @@
-    curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - && \
-    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" && \
-    apt-get update && \
-    apt-get -y install docker-ce && \
-    apt-get update && apt-get install -y git && \
-    apt-get install -y cmake && \
-    apt-get install -y zip && \
-    git clone https://github.com/google/flatbuffers.git
-WORKDIR /home/ubuntu/software/flatbuffers
-RUN git checkout tags/v1.10.0 && \
-    apt-get update && \
-    apt-get install -y build-essential && \
-    cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release && \
-    make
-WORKDIR /home/ubuntu/software
-RUN apt-get install -y wget
-#RUN wget http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz
-RUN git clone https://github.com/chinthakarukshan/metis.git
-WORKDIR /home/ubuntu/software/metis
-RUN tar -xvf metis-5.1.0.tar.gz
-WORKDIR /home/ubuntu/software/metis/metis-5.1.0
-RUN sed -i '/#define IDXTYPEWIDTH 32/c\#define IDXTYPEWIDTH 64' include/metis.h && \
-    make config shared=1 cc=gcc-7 && \
-    make install
-WORKDIR /home/ubuntu/software
-RUN git clone https://github.com/gabime/spdlog.git && \
-    apt-get install -y sqlite3 && \
-    apt-get install -y libsqlite3-dev && \
-    apt install -y librdkafka-dev && \
-    apt-get update && \
-    apt-get install -y libboost-all-dev
-WORKDIR /home/ubuntu/software
-RUN git clone https://github.com/mfontanini/cppkafka.git && \
-    apt-get install -y libssl-dev
-WORKDIR /home/ubuntu/software/cppkafka
-RUN mkdir build
+
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y apt-transport-https
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y curl gnupg2 ca-certificates software-properties-common nlohmann-json3-dev
+
+RUN apt-get install --no-install-recommends -y git cmake build-essential gcc-9 g++-9 sqlite3 libsqlite3-dev libssl-dev librdkafka-dev libboost-all-dev libtool libxerces-c-dev libflatbuffers-dev
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 10
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 20
+
+RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 10
+RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 20
+
+RUN add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get install --no-install-recommends -y python3.11-dev
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+RUN apt-get install --no-install-recommends -y libjsoncpp-dev libspdlog-dev pigz
+RUN python3.11 get-pip.py
+
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y docker-ce-cli
+
+RUN git clone --single-branch --depth 1 https://github.com/mfontanini/cppkafka.git
```

Comment:
Isn't it better to bring the 3,4,5 lines above this line? i.e.
```
WORKDIR /home/ubuntu
RUN mkdir software
WORKDIR /home/ubuntu/software
```
It is confusing where this repository going to be cloned.

Reply:
Yes, changed accordingly.